src/main/java/com/redhat/lightblue/camel/LightblueScheduledPollConsumer.java

```diff
@@ -21,21 +22,29 @@ public LightblueScheduledPollConsumer(LightblueScheduledPollEndpoint endpoint, P
     @Override
     protected int poll() throws Exception {
         Exchange exchange = endpoint.createExchange();
+        LightblueResponse response = null;
 
-        // create a message body
-        LightblueResponse response = endpoint.getLightblueClient().data(endpoint.getLightbluePollingRequest());
-        exchange.getIn().setBody(response);
+        try {
+            response = endpoint.getLightblueClient().data(endpoint.getLightbluePollingRequest());
+            if (response.hasError()) {
+                throw new RuntimeException(response.getText());
```

Comment:
Why not just set the exchange exception here rather than throwing this exception only to have it caught a few lines down.